src/inmemory_cache.rs

```diff
@@ -0,0 +1,136 @@
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+
+use crate::{
+    block::Block, db_state::SsTableId, filter::BloomFilter, flatbuffer_types::SsTableIndexOwned,
+};
+
+#[derive(Clone)]
+pub(crate) enum CachedBlock {
+    Block(Arc<Block>),
+    #[allow(dead_code)]
+    Index(Arc<SsTableIndexOwned>),
+    #[allow(dead_code)]
+    Filter(Arc<BloomFilter>),
+}
+
+#[derive(Clone, Copy)]
+pub enum CacheType {
+    Moka,
+}
+
+#[derive(Clone, Copy)]
+pub struct BlockCacheOptions {
+    pub max_capacity: u64,
+    pub cached_block_size: u32,
+    pub time_to_live: Option<Duration>,
+    pub time_to_idle: Option<Duration>,
+    pub cache_type: CacheType,
+}
+
+impl Default for BlockCacheOptions {
+    fn default() -> Self {
+        Self {
+            max_capacity: 64 * 1024 * 1024, // 64MB default max capacity
+            cached_block_size: 32,          // 32 bytes default
+            time_to_live: None,
+            time_to_idle: None,
+            cache_type: CacheType::Moka, // Default to Moka cache
+        }
+    }
+}
+
+#[async_trait]
+pub(crate) trait BlockCache: Send + Sync + 'static {
```

Comment:
One last nit: I feel like this should just be named `InMemoryCache` or something since it's no longer _just_ a block cache--it can cache filters/indexes, too.

Reply:
Same with the `BlockCacheOption`. Maybe `InMemoryCacheOption` or something.